hub/core/storage/provider.py

```diff
@@ -121,7 +124,25 @@ def __len__(self):
             int: the number of files present inside the root.
         """
 
+    def enable_readonly(self):
+        """Enables read-only mode for the provider."""
+        self.read_only = True
+
+    def disable_readonly(self):
```

Comment:
I don't really see a usecase for this. I don't think we'll allow users to switch modes while the dataset is open anyway

Reply:
Yeah, I kinda just had it as the mirror to enable. I'll remove it